### ISSUE TITLE
Add more dask tests

### ISSUE THREAD

**Issue by scottyhq** [pangeo-data/pangeo-stacks-dev#25]
Since we're moving to dask-gateway for default dask configuration it might be time to revisit old tests setup to ensure dask-kubernetes was properly setup. 

I also commented out the `client.get_versions(check=True)` test because it causing CI to run indefinitely. @TomAugspurger @jhamman - would greatly appreciate your suggestions / PRs to help with this:

https://github.com/pangeo-data/pangeo-stacks-dev/blob/157f3f2fcc12345abddf459d9baaa6c2f160e4ba/tests/test_all.py#L27-L45

**Comment by TomAugspurger** [comment]
What all are we hoping to test here? It seems most things should be outside of pangeo-stacks, so there may not be much to test?

**Comment by scottyhq** [comment]
> What all are we hoping to test here? It seems most things should be outside of pangeo-stacks, so there may not be much to test? 

Yeah, maybe additional ones aren't needed but if we want to carry over the `versions = client.get_versions(check=True)` test or some variant of it, we'll have to figure out why it hangs on CI. I'd also be okay simply removing it and adding additional tests as-needed. 

It seems like the ultimate test is ensuring these images run successfully on binderhub and jupyterhub. The binderbot rendering gallery notebooks may accomplish this for various examples
https://github.com/pangeo-data/pangeo-stacks-dev/issues/18

I was thinking we could also add a CI workflow here that 1) launches pangeo/pangeo-image:master on binderhub and 2) runs a minimal dask-gateway test script on the cluster. Thoughts?

**Comment by TomAugspurger** [comment]
> I'd also be okay simply removing it and adding additional tests as-needed.

I think this may be OK. Recent versions of distributed check package versions on startup.

> I was thinking we could also add a CI workflow here that 1) launches pangeo/pangeo-image:master on binderhub and 2) runs a minimal dask-gateway test script on the cluster. Thoughts?

Seems reasonable. You might be able to reuse binderbot for this. I do worry a bit about that being flaky though.

**Comment by scottyhq** [comment]
@TomAugspurger - given the recent dashboard issues is it worth adding a few tests here for dask-gateway that can run locally (e.g. https://github.com/dask/dask-gateway/issues/261#issuecomment-622034788) ?

**Comment by TomAugspurger** [comment]
Yeah that wouldn't hurt. I'm not sure how best to write those tests though. I had some issues in distributed making a simple `get` to the dashboard URL work correctly.
